src/dispatch/Dispatch.ts

```diff
@@ -142,9 +147,22 @@ export class Dispatch {
     public startDispatchTimer() {
         document.addEventListener(
             'visibilitychange',
-            this.dispatchBeaconFailSilent
+            // The page is moving to the hidden state, which means it may be
+            // unloaded. The sendBeacon API would typically be used in this
+            // case. However, ad-blockers prevent sendBeacon from functioning.
+            // We therefore have two bad options:
+            //
+            // (1) Use sendBeacon and accept missing data when ad blockers are
+            //     used and the page loses visibility
+            // (2) Use fetch and accept missing data when the page is unloaded
+            //     before fetch completes
+            //
+            // A third option is to send both, however this would increase
+            // bandwitch and require deduping server side.
+            this.config.useBeacon
+                ? this.dispatchBeaconFailSilent
+                : this.dispatchFetchFailSilent
         );
-        document.addEventListener('pagehide', this.dispatchBeaconFailSilent);
```

Comment:
question: Are we removing this since `visibilitychange` implies `pagehide`?

Reply:
If so, should we add a unit test case to verify `pagehide` is handled properly as intended?

Reply:
That was the idea, although as I look at the compatibility tables, `pagehide` is supported in older browser versions which the web client technically supports, so perhaps we should keep the event listener for `pagehide`.